pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,384 @@
+            """,
+            set_process=lambda v: Decimal(assume_units(v, ureg.V).m_as(ureg.V)).to_eng_string(
+            ),  # send the value as V to the device
+            get_process=lambda v: ureg.Quantity(v, ureg.V),  # convert to quantity
+            validator=strict_discrete_set,
+            values=VOLTAGE_RANGE_VALUES
+        )
+
+        coupling = Instrument.control(
+            "CONF:CHAN:COUP? (@{ch})",
+            "CONF:CHAN:COUP (@{ch}),%s",
+            """Comtrol channel coupling (AC|DC).""",
+            validator=strict_discrete_set,
+            values=COUPLING_VALUES
+        )
+
+        filter = Instrument.control(
+            "CONF:CHAN:FILT? (@{ch})",
+            "CONF:CHAN:FILT (@{ch}),%s",
+            """Control Filter for this channel (LP_20_MHZ, LP_2_MHZ, LP_200_KHZ).""",
+            validator=strict_discrete_set,
+            values=FILTER_VALUES
+        )
+
+        def _read_data_block(self):
+            # Data block has header of form '#<n><n digits>',
+            # where <n>specifies the number of digits used to indicate the size of the block,
+            # and the <n digits> indicate the block size in bytes.
+            digits = self.read_bytes(2).decode()
+            if not digits.startswith('#'):
+                raise Error()
+            bytes = int(self.read_bytes(int(digits[1])).decode())
+            # Read specified number of bytes + newline terminator
+            buf = memoryview(self.read_bytes(bytes+1))
+            # Return view that strips off newline to contain only desired bytes
+            return buf[:-1]
+
+        @property
+        def voltage(self):
+            """Get voltage measurements for this channel (array of float in V)."""
+            self.write(f'FETC:WAV:VOLT? (@{self.id})')
+            data = self._read_data_block()
+            return ureg.Quantity(np.frombuffer(data, dtype='>f4'), ureg.V)
+
+        @property
+        def adc(self):
+            """
+            Get raw ADC measurements for this channel in counts (int from -32767 to +32767)
+            in current voltage range.
+            """
+            self.write(f'FETC:WAV:ADC? (@{self.id})')
+            data = self._read_data_block()
+            return np.frombuffer(data, dtype='>i2')
+
+    channels = Instrument.ChannelCreator(DigitizerChannel, (1, 2, 3, 4))
+
+    display = Instrument.control(
+        'DISP:TEXT?',
+        'DISP:TEXT \"%s\"',
+        """Control Display text on screen, up to 12 characters.""",
+        get_process=lambda v: v.strip('\"')
+    )
+
+    def clear_display(self):
+        return self.write('DISP:TEXT:CLE')
+
+    arm_source = Instrument.control(
+        "CONF:ARM:SOUR?",
+        "CONF:ARM:SOUR %s",
+        """Set the source used to arm the digitizer (IMMediate|EXTernal|SOFTware|TIMer).""",
+        validator=strict_discrete_set,
+        values=['IMM', 'SOFT', 'EXT', 'TIM']
```

Comment:
The way string `values` are used is not consistent across instruments, and there isn't guidance in the pymeasure docs about it, but I think the abbreviated commands should be mapped to the lowercase word for which they stand for e.g.
```
      values={'immediate': 'IMM', 'software': 'SOFT', 'external': 'EXT','timer': 'TIM'],
      map_values=True
```

That allows procedure code that looks like ` inst.arm_source = 'software' ` which brings more clarity at a glance when a user is trying to understand the procedure.

This shouldn't hold up this PR, it can be a post for discussion or an opened issue, just wondering what you think.

Reply:
Does the device understand the long (lower case) form? And mixed case?

I think it is good, if the property accepts the long form, as it is more expressive than "Arm_source =" IMM"".